### PR TITLE
Fourier expansion layer and generalize cutoff polynomial

### PR DESCRIPTION
**Pull request by lbluque**
- Implements a `FourierExpansion` basis layer.
- Generalized `polynomial_cutoff` to allow setting the min exponent as well. I left the default exponent = 3 to keep the behavior unchanged.
- Added/modified tests

### CODE REVIEW

File: matgl/utils/cutoff.py [hunk 1]
```diff
@@ -7,17 +7,27 @@
...
+        exponent (int): minimum exponent of the polynomial. Default is 5.
+            The polynomial will include terms of order exponent, exponent + 1, exponent + 2.
 
-    Returns: polynomial cutoff functions
-
+    Returns: polynomial cutoff function
     """
+    a = -(exponent + 1) * (exponent + 2) / 2
+    b = exponent * (exponent + 2)
+    c = -exponent * (exponent + 1) / 2
```
Comment (janosh) [inline matgl/utils/cutoff.py:26]: Are these single-letter var names to match the paper? If not, maybe there are more descriptive names?

File: matgl/utils/cutoff.py [hunk 2]
```diff
@@ -7,17 +7,27 @@
...
+    """Envelope polynomial function that ensures a smooth cutoff.
+
+    Ensures first and second derivative vanish at cuttoff. As described in:
+        https://arxiv.org/abs/2003.03123
+
     Args:
         r (torch.tensor): radius distance tensor
         cutoff (float): cutoff distance.
+        exponent (int): minimum exponent of the polynomial. Default is 5.
+            The polynomial will include terms of order exponent, exponent + 1, exponent + 2.
```
Comment (janosh) [inline matgl/utils/cutoff.py:20]: Default is 3

File: matgl/utils/cutoff.py [hunk 3]
```diff
@@ -7,17 +7,27 @@
 import torch
 
 
-def polynomial_cutoff(r, cutoff: float):
-    """Polynomial cutoff function
+def polynomial_cutoff(r: torch.tensor, cutoff: float, exponent: int = 3) -> torch.tensor:
```
Comment (janosh) [inline matgl/utils/cutoff.py:10]:
```suggestion
def polynomial_cutoff(r: torch.Tensor, cutoff: float, exponent: int = 3) -> torch.Tensor:
```

File: matgl/utils/cutoff.py (hunk 2 above)
Comment (lbluque) [reply matgl/utils/cutoff.py:20]: good catch! will fix

**Comment by shyuep** [comment]
I suggest you merge the main branch first. I have already fixed many of the typing issues previously.

**Comment by shyuep** [comment]
Thanks.

**Pull request merged**